### PR TITLE
CLOUDP-212880: local dev - add support for auth and --bindIpAll flag

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes
* add support for auth for local deployments
* add --bindIpAll flag, which allows running local dev inside a docker image

_Jira ticket:_ CLOUDP-212880

[e2e test](https://spruce.mongodb.com/task/mongodb_atlas_cli_master_e2e_required_atlas_deployments_local_e2e_patch_2345231d78e89a06918a52ed5a4067bf7a43e99d_656492205623436eb5e65118_23_11_27_12_57_08/tests?execution=0&sortBy=STATUS&sortDir=ASC)

### CODE REVIEW

File: internal/cli/atlas/deployments/setup.go [hunk 1]
```diff
@@ -769,6 +771,8 @@ func SetupBuilder() *cobra.Command {
 
 	// Local only
 	cmd.Flags().IntVar(&opts.Port, flag.Port, 0, usage.MongodPort)
+	cmd.Flags().BoolVar(&opts.bindAllIPs, flag.BindAllIPs, false, usage.BindAllIPs)
+	_ = cmd.Flags().MarkHidden(flag.BindAllIPs)
```
Comment (wtrocki) [inline internal/cli/atlas/deployments/setup.go:775]: Q: Is this hidden for an initial period or would it be always hidden?

File: internal/podman/client.go [hunk 2]
```diff
@@ -360,7 +361,11 @@ func (o *client) RunContainer(ctx context.Context, opts RunContainerOpts) ([]byt
 	}
 
 	for hostPort, containerPort := range opts.Ports {
-		arg = append(arg, "-p", "127.0.0.1:"+strconv.Itoa(hostPort)+":"+strconv.Itoa(containerPort))
+		portMapping := strconv.Itoa(hostPort) + ":" + strconv.Itoa(containerPort)
+		if !opts.BindAllIPs {
+			portMapping = "127.0.0.1:" + portMapping
+		}
+		arg = append(arg, "-p", portMapping)
```
Comment (wtrocki) [inline internal/podman/client.go:368]: [nit] better redability
```suggestion
                if opts.BindAllIPs {
		      portMapping := strconv.Itoa(hostPort) + ":" + strconv.Itoa(containerPort)
		} else {
		      portMapping = "127.0.0.1:" +  strconv.Itoa(containerPort)
		}
		arg = append(arg, "-p", portMapping)
```

Comment (wtrocki) [inline internal/podman/client.go:368]: [nit] better readability
```suggestion
                if opts.BindAllIPs {
		      portMapping := strconv.Itoa(hostPort) + ":" + strconv.Itoa(containerPort)
		} else {
		      portMapping = "127.0.0.1:" +  strconv.Itoa(containerPort)
		}
		arg = append(arg, "-p", portMapping)
```

Comment (wtrocki) [reply internal/podman/client.go:368]: Actually existing logic is better. Good to ignore

**Review by wtrocki — APPROVED** [review]

**Review by gssbzn — DISMISSED** [review]
would this make it internet accessible?

File: internal/cli/atlas/deployments/setup.go (hunk 1 above)
Comment (tibulca) [reply internal/cli/atlas/deployments/setup.go]: change it to visible in the meantime, as auth support was implemented here

**Comment by tibulca** [comment]
> would this make it internet accessible?

Implemented auth for local deployments to mitigate this concern.
--bindIpAll will work only when auth is enabled.

File: internal/cli/atlas/deployments/setup.go [hunk 3]
```diff
@@ -508,12 +553,33 @@ func (opts *SetupOpts) validateDeploymentTypeFlag() error {
...
+	}
+
 	if opts.DeploymentType != "" && !strings.EqualFold(opts.DeploymentType, options.AtlasCluster) && !strings.EqualFold(opts.DeploymentType, options.LocalCluster) {
 		return fmt.Errorf("%w: %s", errInvalidDeploymentType, opts.DeploymentType)
 	}
 
 	return nil
 }
+
+func (opts *SetupOpts) validateBindIPAllFlag() error {
```
Comment (gssbzn) [inline internal/cli/atlas/deployments/setup.go:567]: consider that cobra now supports flags that are required together and flags that are mutually exclusive even for grouped flags https://github.com/spf13/cobra/blob/main/site/content/user_guide.md#flag-groups

**Review by gssbzn — APPROVED** [review]

**Review by sarahsimpers — APPROVED** [review]
Docs copy LGTM

**Pull request merged**